ambry-commons/src/test/java/com.github.ambry.commons/HelixPropertyStoreUtils.java

```diff
@@ -0,0 +1,198 @@
+        latch.countDown();
+      } else {
+        System.err.println("Data is changed but wrong operation type is specified. " + operationType);
+      }
+    }
+
+    @Override
+    public void onDataCreate(String path) {
+      if (operationType.equals(StoreOperationType.CREATE) || operationType.equals(StoreOperationType.WRITE)) {
+        latch.countDown();
+      } else {
+        System.err.println("Data is created but wrong operation type is specified. " + operationType);
+      }
+    }
+
+    @Override
+    public void onDataDelete(String path) {
+      if (operationType.equals(StoreOperationType.DELETE)) {
+        latch.countDown();
+      } else {
+        System.err.println("Data is deleted but wrong operation type is specified. " + operationType);
+      }
+    }
+  }
+
+  /**
+   * The type of store operations.
+   */
+  enum StoreOperationType {
+    CREATE, WRITE, DELETE
+  }
+
+  /**
+   * Delete corresponding {@code ZooKeeper} nodes of a {@link HelixPropertyStore} if exist, specified by
+   * {@link HelixPropertyStoreConfig}.
+   * @param storeConfig The config that specifies the {@link HelixPropertyStore}.
+   * @param storeFactory The factory to get a {@link HelixPropertyStore}.
+   * @throws InterruptedException
+   */
+  static void deleteStoreIfExists(HelixPropertyStoreConfig storeConfig, HelixPropertyStoreFactory storeFactory)
+      throws InterruptedException {
+    HelixStoreOperator storeOperator = new HelixStoreOperator(storeConfig, storeFactory);
+    if (storeOperator.exist("/")) {
+      storeOperator.delete("/");
+    }
+  }
+
+  /**
+   * Get a {@link HelixStoreOperator}.
+   * @param storeConfig The config that specifies the {@link HelixPropertyStore}.
+   * @param storeFactory The factory to get a {@link HelixPropertyStore}.
+   * @return A {@link HelixStoreOperator}.
+   */
+  static HelixStoreOperator getStoreOperator(HelixPropertyStoreConfig storeConfig,
+      HelixPropertyStoreFactory storeFactory) {
+    return new HelixStoreOperator(storeConfig, storeFactory);
+  }
+
+  /**
+   * A util method that generates {@link HelixPropertyStoreConfig}.
+   * @param zkClientConnectString The connect string to connect to a {@code ZooKeeper}.
+   * @param zkClientSessionTimeoutMs Timeout for a zk session.
+   * @param zkClientConnectionTimeoutMs Timeout for a zk connection.
+   * @param storeRootPath The root path of a store in {@code ZooKeeper}.
+   * @param completeAccountMetadataPath The path of a store in {@code ZooKeeper}, which points to the complete
+   *                                    account metadata path.
+   * @param topicPath The path of a store in {@code ZooKeeper}, which points to the topic path used by the
+   *                  {@link HelixNotifier}.
+   * @return {@link HelixPropertyStoreConfig} defined by the arguments.
+   */
+  static HelixPropertyStoreConfig getHelixStoreConfig(String zkClientConnectString, int zkClientSessionTimeoutMs,
```

Comment:
It is. However, later we may want to extract it out to give us the capability to directly write into a helix store.